nexus/networking/src/firewall_rules.rs

```diff
@@ -266,9 +308,43 @@ pub async fn resolve_firewall_rules_for_sled_agent(
             continue;
         }
 
-        let filter_hosts = match &rule.filter_hosts {
-            None => None,
-            Some(hosts) => {
+        // Construct the set of filter hosts from the DB rule.
+        let filter_hosts = match (&rule.filter_hosts, should_apply_allowlist) {
+            // No host filters, but we need to insert the allowlist entries.
+            //
+            // This is the expected case when applying rules for the built-in
+            // Oxide-services VPCs, which do not contain host filters. (See
+            // `nexus_db_queries::fixed_data::vpc_firewall_rule` for those
+            // rules.) If those rules change to include any filter hosts, this
+            // logic needs to change as well.
+            (None, true) => match allowed_ips.as_ref().unwrap() {
```

Comment:
Tiny nit - could we change the match up on line 311 to be on `&allowed_ips` instead of `should_apply_allowlist`? Then this could be

```suggestion
            (None, Some(allowed_ips)) => match allowed_ips {
```

with no need to unwrap.

Reply:
Yeah, that's much nicer, thanks. Done in an upcoming commit.

Reply:
Done in e2cc1c13d